Assets/Scripts/MarchingCubes/Chunk.cs

```diff
@@ -46,13 +51,56 @@ public void Initialize(int index, int resolution, MarchingCubes marchingCubes, T
         GetComponent<MeshRenderer>().material = material;
     }
 
+    private void calculateChunkPosition()
+    {
+        // Extra the chunkindex in terms of x,y,z
+        int mask = 0;
+        for (int i = 0; i < marchingCubes.chunkResolution; i++)
+        {
+            mask += 1 << i;
+        }
+
+        Vector3 chunkIndex = 
+            new Vector3(
+                (index & (mask << (marchingCubes.chunkResolution * 0))) >> (marchingCubes.chunkResolution * 0),
+                (index & (mask << (marchingCubes.chunkResolution * 1))) >> (marchingCubes.chunkResolution * 1),
+                (index & (mask << (marchingCubes.chunkResolution * 2))) >> (marchingCubes.chunkResolution * 2)
+            );
+
+        // Use the chunkindex to calculate the position of the chunk
+        position = -(chunkIndex - 1.5f * Vector3.one) * (marchingCubes.diameter / (1 << (marchingCubes.chunkResolution)));
+    }
+
     private void Update()
```

Comment:
If there is nothing in the Update method, it should be removed since it is still being run every frame for every chunk

Reply:
fixat